target/snitch_cluster/sw/snax/streamer-gemm-conv-simd/src/snax-streamer-gemm-conv-simd-lib.c

```diff
@@ -157,7 +158,7 @@ void set_gemmx_streamer_csr(
     write_csr(1010, (uint32_t)(delta_local_d32 + snrt_l1_next()));
 
     // transpose or not
-    write_csr(1011, 0);
+    write_csr(1011, (transpose_B) << 1 | transpose_A);
```

Comment:
Hi! Sorry for coming late to the party;

This seems to merge the functionality of two different streamers in the same CSR address space, and it feels a bit ad-hoc and not very systematic to me.

While I would totally agree that spending two different registers on just a a 1-bit setting is total overkill, I do think this kind of breaks the addressing "system" we were using before.
By that I mean that previously we would add up more registers for each streamer, limiting the amount of streamers we can implement by the amount of CSRs one has.

If we do it like this, the amount of CSRs is (somewhat arbitrarily) limited to 32?

Furthermore, doesn't this tie the configuration of all streamers together? Even if at some point in the future they could work a bit more autonomously?

Can we instead maybe create separate CSRs for extra options related to low-bit settings and merge those together instead?
Essentially I'm suggesting to take all streamer CSR address maps apart (like before?):
```c
 write_csr(1011, transpose_A);
 write_csr(1012, transpose_B);
```
Or, if later extra options get added:
```c
 write_csr(1011, future_2_bits_for_A  << 3 | future_1bit_for_A << 1 | transpose_A);
 write_csr(1012, future_2_bits_for_B  << 3 | future_1bit_for_B << 1 | transpose_B);
```

Reply:
Ah! So from what I understood with what you said it's something like, for all configurations related to streamer A, they are like in a set of registers for streamer A only. Then, whatever is for B are just for B alone but not mixed in like one register just like what the PR made.

I would agree that it makes more sense and is more organized.

Reply:
I agree with the sentiment here.
I would suggest removing the csr altogether! We already have them! The spatial strides can define the transposition

Currently, they are always programmed like this:
```c
// spatial strides for A
write_csr(996, spatial_a_0 = 8);
write_csr(997, spatial_a_1 = 1);
```


If we program them like this, a transpose should be applied:
```c
// spatial strides for A
write_csr(996, spatial_a_0 = 1);
write_csr(997, spatial_a_1 = 8);
```

If we do something else, an assertion error should be thrown
```c
// spatial strides for A
write_csr(996, spatial_a_0 = 3);
write_csr(997, spatial_a_1 = 8);
```

But this can be also valid:
```c
// spatial strides for A
write_csr(996, spatial_a_0 = 3);
write_csr(997, spatial_a_1 = 8);
```

I think the rules can be:

- one of the spatial strides should be 1, the other a multiple of 8
- if the first spatial stride is 1, apply transpose, otherwise not

Reply:
@jorendumoulin we can't achieve transpose through this way. The data granularity is 64bit in each TCDM bank.

Reply:
>So from what I understood ...

You understood correctly @rgantonio !

> I would suggest removing the csr altogether! We already have them!

Even better!

Reply:
> @jorendumoulin we can't achieve transpose through this way. The data granularity is 64bit in each TCDM bank.

Ah yes so the last case is not valid, but the first two should be enough, no?

Reply:
@JosseVanDelm we will refactor the csr mapping with the new streamer next week. We can discuss more about the arrangement of the csrs.

Reply:
Wait no it should be valid the strides are bytes

Reply:
@jorendumoulin currently, the spatial_a_0 must be 1 to be aligned with the 64bit data width of TCDM even in the narrow port.

Reply:
Thanks <3 @xiaoling-yi !